packages/vite/src/streaming/createReactStreamingHandler.ts

```diff
@@ -17,9 +17,22 @@ import { MiddlewareResponse } from '../middleware/MiddlewareResponse.js'
 import type { Middleware } from '../middleware/types.js'
 import { makeFilePath } from '../utils.js'
 
+import type { ServerEntryType } from './streamHelpers.js'
 import { reactRenderToStreamResponse } from './streamHelpers.js'
 import { loadAndRunRouteHooks } from './triggerRouteHooks.js'
 
+export type EntryServer =
+  | {
+      middleware?: Middleware
```

Comment:
This is now registerMiddleware :)

Reply:
Hmm, something got lost somewhere when I pulled this change over from another branch of mine... Let's see what happened

Reply:
Thanks Danny! That took some code shuffling to fix 😅
Would not have found that without your comment 🙏